src/renderer/viz/expressions/ramp.js

```diff
@@ -135,6 +143,262 @@ export default class Ramp extends BaseExpression {
+     *   // legend = [
+     *   //   { name: 'numeric', values: [// rgba color, [0, 25 ]] },
+     *   //   { name: 'numeric', values: [// rgba color, [25, 50 ]] }
+     *   //   { name: 'numeric', values: [// rgba color, [50, 75 ]] }
+     *   //   { name: 'numeric', values: [// rgba color, [75, 100 ]] }
+     *   // ]
+     * });
+     * @memberof carto.expressions.Ramp
+     * @name getLegend
+     * @instance
+     * @api
+     */
+    getLegend (options) {
+        const config = Object.assign(DEFAULT_OPTIONS, options);
+
+        if (config.samples > MAX_SAMPLES) {
+            throw new Error(`The maximum number of samples for a legend is ${MAX_SAMPLES}`);
+        }
+
+        if (this.input.isA(Linear)) {
+            return this._getLegendLinear(this.input, config);
+        }
+
+        if (this.input.type === inputTypes.CATEGORY) {
+            return this.input.list
+                ? this._getLegendList(config)
+                : this._getLeyendCategories(config);
+        }
+    }
+
+    _getLegendLinear (input, config) {
+        const firstChild = input[input.childrenNames[0]];
+        return firstChild.isA(Property)
+            ? this._getLegendLinearProperty(firstChild, config)
+            : this._getLegendLinear(firstChild, config);
+    }
+
+    _getLegendLinearProperty (property, config) {
+        const name = property.name;
+        const feature = {};
+        const breakpoints = _getBreakpoints(this._metadata.properties[name], config.samples);
+
+        return breakpoints.map((breakpoint) => {
+            feature[name] = breakpoint[0];
+            const values = [ this.eval(feature), breakpoint ];
+
+            return { name, values };
+        });
+    }
+
+    _getLegendList () {
+        return this.input.list.elems
+            .map(this._getLegendItemValue.bind(this))
+            .filter(legend => legend.values !== null);
+    }
+
+    _getLeyendCategories (config) {
+        return this.input.categories
+            .map((category, index) => { return this._getLegendCategoryValue(category, index, config); })
+            .filter(legend => legend.values !== null);
+    }
+
+    _getLegendItemValue (category, index) {
+        const value = this._getRampValueByIndex(index);
+
+        return {
+            name: this._getCategoryName(category.expr),
+            values: value ? [ value ] : null
+        };
+    }
+
+    _getLegendCategoryValue (category, index, config) {
+        const value = this._getRampValueByIndex(index);
+
+        if (this.input.isA(Top) && this.input.numBuckets === index) {
+            return {
+                name: config.defaultOthers || DEFAULT_OTHERS_NAME,
+                values: value ? [ value ] : null
+            };
+        }
+
+        return {
+            name: this._getCategoryName(category.name),
+            values: value ? [ value ] : null
+        };
+    }
+
+    _getRampValueByIndex (index, config) {
+        if (this.palette.type === paletteTypes.IMAGE) {
```

Comment:
I see a similar problem here too, since `ramp` already has an `eval` method, we could use that